languages/python/reference/exercise-concepts/robot-simulator.md

```diff
@@ -0,0 +1,62 @@
+  
+- instance methods: Tests for this exercises require one or more instance methods that will take in a set of starting coordinates and a bearing and then accept a seiries of instructions that "move" the instance to a new set of coordinates and bearing.
+
+- function decorator: a higer-order function that takes another function as an argument. The "decorating" function extends the behavior of the "decorated" function without explicitly modifying it (i.e. it _wraps_ or _decorates_ it). Called in python by using the `@` symbol and the function name ahead of the function being decorated.  The example uses pythons built-in `property()` as a decorator (`@property`) to return a "compound" read-only instance property made up of two seperate instance properties.
+
+- higher-order functions:  a function that takes one or more other functions as arguments, _returning_ a function as its retrun value.  The example uses the built-in `property()` as a higher-order function through `@property`.
+
+- `property()` built-in:  a function that returns a property attribute.  When used as a decorator, this transforms the passed-in method into a _getter_ method for read-only attribute with the same name and docstring.
+
+- assignment:  the example uses assignment for all the instance properties and `instructions` dictionary.
+
+- instance properties: This exercise rquires one or more instance properties to persist passed in data.
+
+- mutability: In the example, knowing there are no protected or private properties in python and so conciously mutating `self.x`, `self.y` and `self.compass` through the called instance methods.
+
+- method arguments: the example `__init__` method needs to take `self`, direction, x, and y (coordinates) as arguments.  It also uses `self` and `commands` (a string) for arguments to the `move()` method.
+- default arguments:  Pre-setting a function argument value to protect against it not being passed..  The example uses `direction = NORTH` and `x=0, y=0` to ensure those values for a robot even if they are not passed in initally.
+
+- dictionaries/mapping type: the example uses a dictionary to map paassed in move arguemtns to methods that perform the moves.  The example also uses a dictionary/mapping created by calling `str.maketrans()`.
+
+- bracket notation in context of dictionaries: the example uses passed in move arguments as keys via bracket notation to look up corresponding methods for moving the robot in the _insructions_ dictionary. 
+
+- iteration/iterators/iterating: the example uses a `for loop` to iterate through the passed in command string.
+
+- composition:  adding functionality from a class by incorporating and instance of that class in a class you are creating.  The example cureates a `robot` by instatiating a `compass`, and assiging it to the `self`.compass property of `robot`.
```

Comment:
Spelling: "cureates" -> "creates", "instatiating" -> "instantiating", "assiging" -> "assigning". Also it should be "an instance" not "and instance". This is another example where attribute may be more clear than property, since `@property` has special meaning in Python.

Reply:
Here's the re-phrase:

> - composition:  adding functionality from a class by incorporating an instance of that class in a class you are creating.  The example creates a `robot` by instantiating a `compass` and assigning it to the `self`.compass attribute of `robot`.